radix-transaction-scenarios/src/scenario.rs

```diff
@@ -375,25 +373,20 @@ pub struct ScenarioMetadata {
     /// - This is used in Node genesis to specify which scenarios should be run
     /// - This should be spaceless as it will be used for a file path
     pub logical_name: &'static str,
+    /// The minimal protocol version required to successfully run this scenario.
+    pub protocol_min_requirement: ProtocolVersion,
+    /// If set, this will run immediately after this protocol update on a testnet.
+    /// Note that setting this will change the definition of the given protocol update,
+    /// so shouldn't be changed once the protocol update is locked in.
+    pub testnet_run_at: Option<ProtocolVersion>,
```

Comment:
As mentioned in the pr description my vision was to have a validate test to validate all the metadata, I think that would be most flexible.

Also happy with a constructor, but I like the named fields tbh 😄

Reply:
It's pretty much whether we want to do this check at compile-time or at runtime. In the approach that I mentioned if a string literal is provided that makes the constructor fail that is a compile-time error which I think is quite neat! Super big fan of these Rust features so it's always fun to find a chance to use them :D But as you said, this could also be done as a runtime check.

Will keep this as a runtime check since you mentioned that you prefer the named fields. I've added a test for this `validate_the_metadata_of_all_scenarios`.